File: cpp-tiny/lib/Models/BasicStream.cpp


#include "BasicStream.h"

using namespace Tiny;

BasicStream::BasicStream()
{
	stream_id = null;
	name = null;
	description = null;
	date_created = null;
	invite_only = null;
	rendered_description = null;
	is_web_public = null;
	stream_post_policy = null;
	message_retention_days = null;
	history_public_to_subscribers = null;
	first_message_id = null;
	is_announcement_only = null;
}

BasicStream::BasicStream(std::string jsonString)
{
	this->fromJson(jsonString);
}

BasicStream::~BasicStream()
{

}

void
BasicStream::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *stream_idKey = "stream_id";
    
    if(object.has_key(stream_idKey)) 
    {
        bourne::json value = object[stream_idKey];


        

        AnyType* obj = &stream_id;
		obj->fromJson(value.dump());

    }

    const char *nameKey = "name";
    
    if(object.has_key(nameKey)) 
    {
        bourne::json value = object[nameKey];


        

        AnyType* obj = &name;
		obj->fromJson(value.dump());

    }

    const char *descriptionKey = "description";
    
    if(object.has_key(descriptionKey)) 
    {
        bourne::json value = object[descriptionKey];


        

        AnyType* obj = &description;
		obj->fromJson(value.dump());

    }

    const char *date_createdKey = "date_created";
    
    if(object.has_key(date_createdKey)) 
    {
        bourne::json value = object[date_createdKey];


        

        AnyType* obj = &date_created;
		obj->fromJson(value.dump());

    }

    const char *invite_onlyKey = "invite_only";
    
    if(object.has_key(invite_onlyKey)) 
    {
        bourne::json value = object[invite_onlyKey];


        

        AnyType* obj = &invite_only;
		obj->fromJson(value.dump());

    }

    const char *rendered_descriptionKey = "rendered_description";
    
    if(object.has_key(rendered_descriptionKey)) 
    {
        bourne::json value = object[rendered_descriptionKey];


        

        AnyType* obj = &rendered_description;
		obj->fromJson(value.dump());

    }

    const char *is_web_publicKey = "is_web_public";
    
    if(object.has_key(is_web_publicKey)) 
    {
        bourne::json value = object[is_web_publicKey];


        

        AnyType* obj = &is_web_public;
		obj->fromJson(value.dump());

    }

    const char *stream_post_policyKey = "stream_post_policy";
    
    if(object.has_key(stream_post_policyKey)) 
    {
        bourne::json value = object[stream_post_policyKey];


        

        AnyType* obj = &stream_post_policy;
		obj->fromJson(value.dump());

    }

    const char *message_retention_daysKey = "message_retention_days";
    
    if(object.has_key(message_retention_daysKey)) 
    {
        bourne::json value = object[message_retention_daysKey];


        

        AnyType* obj = &message_retention_days;
		obj->fromJson(value.dump());

    }

    const char *history_public_to_subscribersKey = "history_public_to_subscribers";
    
    if(object.has_key(history_public_to_subscribersKey)) 
    {
        bourne::json value = object[history_public_to_subscribersKey];


        

        AnyType* obj = &history_public_to_subscribers;
		obj->fromJson(value.dump());

    }

    const char *first_message_idKey = "first_message_id";
    
    if(object.has_key(first_message_idKey)) 
    {
        bourne::json value = object[first_message_idKey];


        

        AnyType* obj = &first_message_id;
		obj->fromJson(value.dump());

    }

    const char *is_announcement_onlyKey = "is_announcement_only";
    
    if(object.has_key(is_announcement_onlyKey)) 
    {
        bourne::json value = object[is_announcement_onlyKey];


        

        AnyType* obj = &is_announcement_only;
		obj->fromJson(value.dump());

    }


}

bourne::json
BasicStream::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["stream_id"] = getStreamId().toJson();

    




	object["name"] = getName().toJson();

    




	object["description"] = getDescription().toJson();

    




	object["date_created"] = getDateCreated().toJson();

    




	object["invite_only"] = getInviteOnly().toJson();

    




	object["rendered_description"] = getRenderedDescription().toJson();

    




	object["is_web_public"] = getIsWebPublic().toJson();

    




	object["stream_post_policy"] = getStreamPostPolicy().toJson();

    




	object["message_retention_days"] = getMessageRetentionDays().toJson();

    




	object["history_public_to_subscribers"] = getHistoryPublicToSubscribers().toJson();

    




	object["first_message_id"] = getFirstMessageId().toJson();

    




	object["is_announcement_only"] = getIsAnnouncementOnly().toJson();


    return object;

}

AnyType
BasicStream::getStreamId()
{
	return stream_id;
}

void
BasicStream::setStreamId(AnyType  stream_id)
{
	this->stream_id = stream_id;
}

AnyType
BasicStream::getName()
{
	return name;
}

void
BasicStream::setName(AnyType  name)
{
	this->name = name;
}

AnyType
BasicStream::getDescription()
{
	return description;
}

void
BasicStream::setDescription(AnyType  description)
{
	this->description = description;
}

AnyType
BasicStream::getDateCreated()
{
	return date_created;
}

void
BasicStream::setDateCreated(AnyType  date_created)
{
	this->date_created = date_created;
}

AnyType
BasicStream::getInviteOnly()
{
	return invite_only;
}

void
BasicStream::setInviteOnly(AnyType  invite_only)
{
	this->invite_only = invite_only;
}

AnyType
BasicStream::getRenderedDescription()
{
	return rendered_description;
}

void
BasicStream::setRenderedDescription(AnyType  rendered_description)
{
	this->rendered_description = rendered_description;
}

AnyType
BasicStream::getIsWebPublic()
{
	return is_web_public;
}

void
BasicStream::setIsWebPublic(AnyType  is_web_public)
{
	this->is_web_public = is_web_public;
}

AnyType
BasicStream::getStreamPostPolicy()
{
	return stream_post_policy;
}

void
BasicStream::setStreamPostPolicy(AnyType  stream_post_policy)
{
	this->stream_post_policy = stream_post_policy;
}

AnyType
BasicStream::getMessageRetentionDays()
{
	return message_retention_days;
}

void
BasicStream::setMessageRetentionDays(AnyType  message_retention_days)
{
	this->message_retention_days = message_retention_days;
}

AnyType
BasicStream::getHistoryPublicToSubscribers()
{
	return history_public_to_subscribers;
}

void
BasicStream::setHistoryPublicToSubscribers(AnyType  history_public_to_subscribers)
{
	this->history_public_to_subscribers = history_public_to_subscribers;
}

AnyType
BasicStream::getFirstMessageId()
{
	return first_message_id;
}

void
BasicStream::setFirstMessageId(AnyType  first_message_id)
{
	this->first_message_id = first_message_id;
}

AnyType
BasicStream::getIsAnnouncementOnly()
{
	return is_announcement_only;
}

void
BasicStream::setIsAnnouncementOnly(AnyType  is_announcement_only)
{
	this->is_announcement_only = is_announcement_only;
}

File: java-pkmst/src/main/java/com/prokarma/pkmst/model/BasicStream.java
package com.prokarma.pkmst.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.prokarma.pkmst.model.BasicStreamAllOf;
import com.prokarma.pkmst.model.BasicStreamBase;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
/**
 * Response class to be returned by Api
 * @author pkmst
 *
 */
/**
 * BasicStream
 */

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPKMSTServerCodegen", date = "2021-08-08T21:19:29.098714Z[Etc/UTC]")
public class BasicStream   {
  @JsonProperty("stream_id")
  private Object streamId = null;

  @JsonProperty("name")
  private Object name = null;

  @JsonProperty("description")
  private Object description = null;

  @JsonProperty("date_created")
  private Object dateCreated = null;

  @JsonProperty("invite_only")
  private Object inviteOnly = null;

  @JsonProperty("rendered_description")
  private Object renderedDescription = null;

  @JsonProperty("is_web_public")
  private Object isWebPublic = null;

  @JsonProperty("stream_post_policy")
  private Object streamPostPolicy = null;

  @JsonProperty("message_retention_days")
  private Object messageRetentionDays = null;

  @JsonProperty("history_public_to_subscribers")
  private Object historyPublicToSubscribers = null;

  @JsonProperty("first_message_id")
  private Object firstMessageId = null;

  @JsonProperty("is_announcement_only")
  private Object isAnnouncementOnly = null;

  public BasicStream streamId(Object streamId) {
    this.streamId = streamId;
    return this;
  }

   /**
   * Get streamId
   * @return streamId
  **/
  @ApiModelProperty(value = "")
  public Object getStreamId() {
    return streamId;
  }

  public void setStreamId(Object streamId) {
    this.streamId = streamId;
  }

  public BasicStream name(Object name) {
    this.name = name;
    return this;
  }

   /**
   * Get name
   * @return name
  **/
  @ApiModelProperty(value = "")
  public Object getName() {
    return name;
  }

  public void setName(Object name) {
    this.name = name;
  }

  public BasicStream description(Object description) {
    this.description = description;
    return this;
  }

   /**
   * Get description
   * @return description
  **/
  @ApiModelProperty(value = "")
  public Object getDescription() {
    return description;
  }

  public void setDescription(Object description) {
    this.description = description;
  }

  public BasicStream dateCreated(Object dateCreated) {
    this.dateCreated = dateCreated;
    return this;
  }

   /**
   * Get dateCreated
   * @return dateCreated
  **/
  @ApiModelProperty(value = "")
  public Object getDateCreated() {
    return dateCreated;
  }

  public void setDateCreated(Object dateCreated) {
    this.dateCreated = dateCreated;
  }

  public BasicStream inviteOnly(Object inviteOnly) {
    this.inviteOnly = inviteOnly;
    return this;
  }

   /**
   * Get inviteOnly
   * @return inviteOnly
  **/
  @ApiModelProperty(value = "")
  public Object getInviteOnly() {
    return inviteOnly;
  }

  public void setInviteOnly(Object inviteOnly) {
    this.inviteOnly = inviteOnly;
  }

  public BasicStream renderedDescription(Object renderedDescription) {
    this.renderedDescription = renderedDescription;
    return this;
  }

   /**
   * Get renderedDescription
   * @return renderedDescription
  **/
  @ApiModelProperty(value = "")
  public Object getRenderedDescription() {
    return renderedDescription;
  }

  public void setRenderedDescription(Object renderedDescription) {
    this.renderedDescription = renderedDescription;
  }

  public BasicStream isWebPublic(Object isWebPublic) {
    this.isWebPublic = isWebPublic;
    return this;
  }

   /**
   * Get isWebPublic
   * @return isWebPublic
  **/
  @ApiModelProperty(value = "")
  public Object getIsWebPublic() {
    return isWebPublic;
  }

  public void setIsWebPublic(Object isWebPublic) {
    this.isWebPublic = isWebPublic;
  }

  public BasicStream streamPostPolicy(Object streamPostPolicy) {
    this.streamPostPolicy = streamPostPolicy;
    return this;
  }

   /**
   * Get streamPostPolicy
   * @return streamPostPolicy
  **/
  @ApiModelProperty(value = "")
  public Object getStreamPostPolicy() {
    return streamPostPolicy;
  }

  public void setStreamPostPolicy(Object streamPostPolicy) {
    this.streamPostPolicy = streamPostPolicy;
  }

  public BasicStream messageRetentionDays(Object messageRetentionDays) {
    this.messageRetentionDays = messageRetentionDays;
    return this;
  }

   /**
   * Get messageRetentionDays
   * @return messageRetentionDays
  **/
  @ApiModelProperty(value = "")
  public Object getMessageRetentionDays() {
    return messageRetentionDays;
  }

  public void setMessageRetentionDays(Object messageRetentionDays) {
    this.messageRetentionDays = messageRetentionDays;
  }

  public BasicStream historyPublicToSubscribers(Object historyPublicToSubscribers) {
    this.historyPublicToSubscribers = historyPublicToSubscribers;
    return this;
  }

   /**
   * Get historyPublicToSubscribers
   * @return historyPublicToSubscribers
  **/
  @ApiModelProperty(value = "")
  public Object getHistoryPublicToSubscribers() {
    return historyPublicToSubscribers;
  }

  public void setHistoryPublicToSubscribers(Object historyPublicToSubscribers) {
    this.historyPublicToSubscribers = historyPublicToSubscribers;
  }

  public BasicStream firstMessageId(Object firstMessageId) {
    this.firstMessageId = firstMessageId;
    return this;
  }

   /**
   * Get firstMessageId
   * @return firstMessageId
  **/
  @ApiModelProperty(value = "")
  public Object getFirstMessageId() {
    return firstMessageId;
  }

  public void setFirstMessageId(Object firstMessageId) {
    this.firstMessageId = firstMessageId;
  }

  public BasicStream isAnnouncementOnly(Object isAnnouncementOnly) {
    this.isAnnouncementOnly = isAnnouncementOnly;
    return this;
  }

   /**
   * Get isAnnouncementOnly
   * @return isAnnouncementOnly
  **/
  @ApiModelProperty(value = "")
  public Object getIsAnnouncementOnly() {
    return isAnnouncementOnly;
  }

  public void setIsAnnouncementOnly(Object isAnnouncementOnly) {
    this.isAnnouncementOnly = isAnnouncementOnly;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BasicStream basicStream = (BasicStream) o;
    return Objects.equals(this.streamId, basicStream.streamId) &&
        Objects.equals(this.name, basicStream.name) &&
        Objects.equals(this.description, basicStream.description) &&
        Objects.equals(this.dateCreated, basicStream.dateCreated) &&
        Objects.equals(this.inviteOnly, basicStream.inviteOnly) &&
        Objects.equals(this.renderedDescription, basicStream.renderedDescription) &&
        Objects.equals(this.isWebPublic, basicStream.isWebPublic) &&
        Objects.equals(this.streamPostPolicy, basicStream.streamPostPolicy) &&
        Objects.equals(this.messageRetentionDays, basicStream.messageRetentionDays) &&
        Objects.equals(this.historyPublicToSubscribers, basicStream.historyPublicToSubscribers) &&
        Objects.equals(this.firstMessageId, basicStream.firstMessageId) &&
        Objects.equals(this.isAnnouncementOnly, basicStream.isAnnouncementOnly);
  }

  @Override
  public int hashCode() {
    return Objects.hash(streamId, name, description, dateCreated, inviteOnly, renderedDescription, isWebPublic, streamPostPolicy, messageRetentionDays, historyPublicToSubscribers, firstMessageId, isAnnouncementOnly);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BasicStream {\n");
    
    sb.append("    streamId: ").append(toIndentedString(streamId)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    dateCreated: ").append(toIndentedString(dateCreated)).append("\n");
    sb.append("    inviteOnly: ").append(toIndentedString(inviteOnly)).append("\n");
    sb.append("    renderedDescription: ").append(toIndentedString(renderedDescription)).append("\n");
    sb.append("    isWebPublic: ").append(toIndentedString(isWebPublic)).append("\n");
    sb.append("    streamPostPolicy: ").append(toIndentedString(streamPostPolicy)).append("\n");
    sb.append("    messageRetentionDays: ").append(toIndentedString(messageRetentionDays)).append("\n");
    sb.append("    historyPublicToSubscribers: ").append(toIndentedString(historyPublicToSubscribers)).append("\n");
    sb.append("    firstMessageId: ").append(toIndentedString(firstMessageId)).append("\n");
    sb.append("    isAnnouncementOnly: ").append(toIndentedString(isAnnouncementOnly)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

